### PR TITLE
RHDEVDOCS-4109 - Release Notes Header fixes

### PR DESCRIPTION
**Pull request by libander**
Version(s): main, 4.12, 4.11, 4.10, 4.9, 4.8

Issue: https://issues.redhat.com/browse/RHDEVDOCS-4109

Link to docs preview: https://libander.github.io/openshift-docs/RHDEVDOCS-4069/logging/cluster-logging-release-notes.html

Additional information: Branch name/JIRA mismatch due to z-stream release delay

### CODE REVIEW

File: logging/cluster-logging-release-notes.adoc [hunk 1]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+* Before this update, a prior change to fix dashboard reconciliation introduced a `ownerReferences` field to the resource across namespaces. As a result, both the config map and dashboard were not getting created in the namespace. With this update, the removal of the `ownerReferences` field resolves the issue and the OpenShift Logging dashboard is available in the console. (link:https://issues.redhat.com/browse/LOG-2163[LOG-2163])
+
+* Before this update, changes to the metrics dashboards did not deploy because the `cluster-logging-operator` did not correctly compare existing and desired configmaps containing the dashboard. With this update, the addition of a unique hash value to object labels resolves the issue.	(link:https://issues.redhat.com/browse/LOG-2071[LOG-2071])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:72]:
```suggestion
* Before this update, changes to the metrics dashboards did not deploy because the `cluster-logging-operator` <noun> did not correctly compare existing and modified config maps that contain the dashboard. With this update, the addition of a unique hash value to object labels resolves the issue.	(link:https://issues.redhat.com/browse/LOG-2071[LOG-2071])
```

> ! desire verb
Do not use. Use "want".

File: logging/cluster-logging-release-notes.adoc [hunk 2]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+*	Before this update, a name change of the deployed collector in the 5.3 release caused the logging collector to generate the `FluentdNodeDown` alert. This update resolves the issue by fixing the job name for the Prometheus alert. (link:https://issues.redhat.com/browse/LOG-1918[LOG-1918])
+
+*	Before this update, the log collector was collecting its own logs due to a refactoring of the component name change. This could lead to a potential feedback loop of the collector processing its own log that might result in memory and log message size issues. This update resolves the issue by excluding the collector logs from the collection. (link:https://issues.redhat.com/browse/LOG-1774[LOG-1774])
+
+* Before this update, Elasticsearch generated the error "Unable to create PersistentVolumeClaim due to forbidden: exceeded quota: infra-storage-quota." if the PVC already existed. With this update, Elasticsearch checks for existing PVCs, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2131[LOG-2131])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:86]:
```suggestion
* Before this update, Elasticsearch generated the error `Unable to create PersistentVolumeClaim due to forbidden: exceeded quota: infra-storage-quota.` if the PVC already existed. With this update, Elasticsearch checks for existing PVCs, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2131[LOG-2131])
```

File: logging/cluster-logging-release-notes.adoc [hunk 3]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+* Before this update, the change of the path from which the collector reads container logs caused the collector to forward some records to the wrong indices. With this update, the collector now uses the correct configuration to resolve the issue. (link:https://issues.redhat.com/browse/LOG-2160[LOG-2160])
+
+* Before this update, clusters with a large number of namespaces caused Elasticsearch to stop serving requests because the list of namespaces reached the maximum header size limit. With this update, headers only include a list of namespace names, resolving the issue. (link:https://issues.redhat.com/browse/LOG-1899[LOG-1899])
+
+* Before this update, the *{product-title} Logging* dashboard showed the number of shards 'x' times larger than the actual value when Elasticsearch had 'x' nodes. This issue occurred because it was printing all primary shards for each ES pod and calculating a sum on it, although the output was always for the whole ES cluster. With this update, the number of shards is now correctly calculated. (link:https://issues.redhat.com/browse/LOG-2156[LOG-2156])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:94]:
```suggestion
* Before this update, the *{product-title} Logging* dashboard showed the number of shards 'x' times larger than the actual value when Elasticsearch had 'x' nodes. This issue occurred because it was printing all primary shards for each Elasticsearch pod and calculating a sum on it, although the output was always for the whole Elasticsearch cluster. With this update, the number of shards is now correctly calculated. (link:https://issues.redhat.com/browse/LOG-2156[LOG-2156])
```

File: logging/cluster-logging-release-notes.adoc [hunk 4]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+* Before this update, the secrets "kibana" and "kibana-proxy" were not recreated if they were deleted manually. With this update, the `elasticsearch-operator` will watch the resources and automatically recreate them if deleted.	(link:https://issues.redhat.com/browse/LOG-2250[LOG-2250])
+
+* Before this update, tuning the buffer chunk size could cause the collector to generate a warning about the chunk size exceeding the byte limit for the event stream. With this update, you can also tune the read line limit, resolving the issue.	(link:https://issues.redhat.com/browse/LOG-2379[LOG-2379])
+
+* Before this update, the logging console link in OpenShift WebConsole was not removed with the ClusterLogging CR. With this update, deleting the CR or uninstalling the Cluster Logging Operator removes the link. (link:https://issues.redhat.com/browse/LOG-2373[LOG-2373])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:100]:
```suggestion
* Before this update, the logging console link in the OpenShift web console was not removed with the ClusterLogging CR. With this update, deleting the CR or uninstalling the Cluster Logging Operator removes the link. (link:https://issues.redhat.com/browse/LOG-2373[LOG-2373])
```

File: logging/cluster-logging-release-notes.adoc [hunk 5]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+* Before this update, a change to the container logs path caused this metric to always be zero with older releases configured with the original path. With this update, the plugin which exposes metrics about collected logs supports reading from either path to resolve the issue. (link:https://issues.redhat.com/browse/LOG-2462[LOG-2462])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:102]: Which metric?

```suggestion
* Before this update, a change to the container logs path caused this metric to always be zero with older releases configured with the original path. With this update, the plug-in that exposes metrics about collected logs supports reading from either path to resolve the issue. (link:https://issues.redhat.com/browse/LOG-2462[LOG-2462])
```
> 🚫  plugin noun, adjective
Do not use. Use "plug-in".

File: logging/cluster-logging-release-notes.adoc [hunk 6]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+//Z-stream Release Notes by Version
+[id="cluster-logging-release-notes-5-3-7"]
+== OpenShift Logging 5.3.7
+This release includes link:https://access.redhat.com/errata/RHSA-2022:2217[RHSA-2022:2217 OpenShift Logging Bug Fix Release 5.3.7]
+
+[id="openshift-logging-5-3-7-bug-fixes"]
+=== Bug fixes
+* Before this update, Linux audit log time parsing relied on an ordinal position of key/value pair. This update changes the parsing to utilize a regex to find the time entry. (https://issues.redhat.com/browse/LOG-2322[LOG-2322])
+
+* Before this update some log forwarder outputs could re-order logs with the same time-stamp. With this update, a sequence number has been added to the log record to order entries that have matching timestamps. (https://issues.redhat.com/browse/LOG-2334[LOG-2334])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:120]:
```suggestion
* Before this update, some log forwarder outputs could re-order logs with the same time-stamp. With this update, a sequence number has been added to the log record to order entries that have matching timestamps. (https://issues.redhat.com/browse/LOG-2334[LOG-2334])
```

File: logging/cluster-logging-release-notes.adoc [hunk 7]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+
+* Before this update, clusters with a large number of namespaces caused Elasticsearch to stop serving requests because the list of namespaces reached the maximum header size limit. With this update, headers only include a list of namespace names, resolving the issue. (https://issues.redhat.com/browse/LOG-2475[LOG-2475])
+
+* Before this update, `system:serviceaccount:openshift-monitoring:prometheus-k8s` had cluster level privileges as a `clusterrole` and `clusterrolebinding`. This update restricts the `serviceaccount` to the `openshift-logging` namespace with a role and rolebinding. (https://issues.redhat.com/browse/LOG-2480[LOG-2480])
+
+* Before this update, the `cluster-logging-operator` utilized cluster scoped roles and bindings to establish permissions for the Prometheus service account to scrape metrics. These permissions were only created when deploying the Operator using the console interface, and were missing when the Operator was deployed from the command line. This fixes the issue by making this role and binding namespace scoped. (https://issues.redhat.com/browse/LOG-1972[LOG-1972])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:541]:
```suggestion
* Before this update, the `cluster-logging-operator` utilized cluster scoped roles and bindings to establish permissions for the Prometheus service account to scrape metrics. These permissions were only created when deploying the Operator using the console interface and were missing when the Operator was deployed from the command line. This fixes the issue by making this role and binding namespace scoped. (https://issues.redhat.com/browse/LOG-1972[LOG-1972])
```

File: logging/cluster-logging-release-notes.adoc [hunk 8]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+* Before this update, defining a toleration with no key and the existing Operator caused the Operator to be unable to complete an upgrade. With this update, this toleration no longer blocks the upgrade from completing. (link:https://issues.redhat.com/browse/LOG-2304[LOG-2304])
+
+[id="cluster-logging-release-notes-5-2-8"]
+== OpenShift Logging 5.2.8
+
+This release includes link:https://access.redhat.com/errata/RHSA-2022:0728[RHSA-2022:0728 OpenShift Logging Bug Fix Release 5.2.8]
+
+[id="openshift-logging-5-2-8-bug-fixes"]
+=== Bug fixes
+* Before this update, if you removed OpenShift Logging from {product-title}, the web console continued displaying a link to the *Logging* page. With this update, removing or uninstalling OpenShift Logging also removes that link.(link:https://issues.redhat.com/browse/LOG-2180[LOG-2180])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:581]:
```suggestion
* Before this update, if you removed OpenShift Logging from {product-title}, the web console continued displaying a link to the *Logging* page. With this update, removing or uninstalling OpenShift Logging also removes that link. (link:https://issues.redhat.com/browse/LOG-2180[LOG-2180])
```

File: logging/cluster-logging-release-notes.adoc [hunk 9]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+====
+
+[id="cluster-logging-release-notes-5-2-7"]
+== OpenShift Logging 5.2.7
+
+This release includes link:https://access.redhat.com/errata/RHBA-2022:0478[RHBA-2022:0478 OpenShift Logging Bug Fix Release 5.2.7]
+
+[id="openshift-logging-5-2-7-bug-fixes"]
+=== Bug fixes
+* Before this update, Elasticsearch pods failed to start after updating with FIPS enabled. With this update, Elasticsearch pods start successfully. (link:https://issues.redhat.com/browse/LOG-2000[LOG-2000])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:601]: Updating what?

File: logging/cluster-logging-release-notes.adoc [hunk 10]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+====
+
+[id="cluster-logging-release-notes-5-2-6"]
+== OpenShift Logging 5.2.6
+
+This release includes link:https://access.redhat.com/errata/RHSA-2022:0230[RHSA-2022:0230 OpenShift Logging Bug Fix Release 5.2.6]
+
+[id="openshift-logging-5-2-6-bug-fixes"]
+=== Bug fixes
+* Before this update, the release did not include a filter change which caused fluentd to crash. With this update, the missing filter has been corrected. (link:https://issues.redhat.com/browse/LOG-2104[LOG-2104])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:628]:
```suggestion
* Before this update, the release did not include a filter change, which caused Fluentd to crash. With this update, the missing filter has been corrected. (link:https://issues.redhat.com/browse/LOG-2104[LOG-2104])
```

File: logging/cluster-logging-release-notes.adoc [hunk 11]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+
+[id="cluster-logging-release-notes-5-2-4"]
+== OpenShift Logging 5.2.4
+
+This release includes link:https://access.redhat.com/errata/RHSA-2021:5127[RHSA-2021:5127 OpenShift Logging Bug Fix Release 5.2.4]
+
+[id="openshift-logging-5-2-4-bug-fixes"]
+=== Bug fixes
+
+* Before this update records shipped via syslog would serialize a ruby hash encoding key/value pairs to contain a '=>' character, as well as replace tabs with "#11".  This update serializes the message correctly as proper JSON. (link:https://issues.redhat.com/browse/LOG-1775[LOG-1775])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:672]:
```suggestion
* Before this update, records shipped via syslog would serialize a ruby hash encoding key/value pairs to contain a '=>' character, as well as replace tabs with "#11".  This update serializes the message correctly as proper JSON. (link:https://issues.redhat.com/browse/LOG-1775[LOG-1775])
```

File: logging/cluster-logging-release-notes.adoc [hunk 12]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+
+[id="cluster-logging-release-notes-5-2-3"]
+== OpenShift Logging 5.2.3
+
+This release includes link:https://access.redhat.com/errata/RHSA-2021:4032[RHSA-2021:4032 OpenShift Logging Bug Fix Release 5.2.3]
+
+[id="openshift-logging-5-2-3-bug-fixes"]
+=== Bug fixes
+
+* Before this update, some alerts did not include a namespace label. This omission doesn't comply with the OpenShift Monitoring Team's guidelines for writing alerting rules in OpenShift. With this update, all the alerts in Elasticsearch Operator include a namespace label and follow all the guidelines for writing alerting rules in OpenShift.(link:https://issues.redhat.com/browse/LOG-1857[LOG-1857])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:754]:
```suggestion
* Before this update, some alerts did not include a namespace label. This omission does not comply with the OpenShift Monitoring Team's guidelines for writing alerting rules in {product-title}. With this update, all the alerts in Elasticsearch Operator include a namespace label and follow all the guidelines for writing alerting rules in {product-title}.(link:https://issues.redhat.com/browse/LOG-1857[LOG-1857])
```

File: logging/cluster-logging-release-notes.adoc [hunk 13]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+
+* Before this update, a regression introduced in a prior release intentionally disabled JSON message parsing. This update re-enables JSON parsing. It also sets the log entry "level" based on the "level" field in parsed JSON message or by using regex to extract a match from a message field. (link:https://issues.redhat.com/browse/LOG-1759[LOG-1759])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:756]:
```suggestion
* Before this update, a regression introduced in a prior release intentionally disabled JSON message parsing. This update re-enables JSON parsing. It also sets the log entry `level` based on the `level` field in parsed JSON message or by using regex to extract a match from a message field. (link:https://issues.redhat.com/browse/LOG-1759[LOG-1759])
```

File: logging/cluster-logging-release-notes.adoc [hunk 14]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+== OpenShift Logging 5.2.2
+
+This release includes link:https://access.redhat.com/errata/RHBA-2021:3747[RHBA-2021:3747 OpenShift Logging Bug Fix Release 5.2.2]
+
+[id="openshift-logging-5-2-2-bug-fixes"]
+=== Bug fixes
+
+* Before this update, the `ClusterLogging` custom resource (CR) applied the value of the `totalLimitSize` field to the Fluentd `total_limit_size` field, even if the required buffer space was not available. With this update, the CR applies the lesser of the two `totalLimitSize` or 'default' values to the Fluentd `total_limit_size` field, resolving the issue.(link:https://issues.redhat.com/browse/LOG-1738[LOG-1738])
+
+* Before this update, a regression introduced in a prior release configuration caused the collector to flush its buffered messages before shutdown, creating a delay the termination and restart of collector Pods. With this update, fluentd no longer flushes buffers at shutdown, resolving the issue. (link:https://issues.redhat.com/browse/LOG-1739[LOG-1739])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:818]:
```suggestion
* Before this update, a regression introduced in a prior release configuration caused the collector to flush its buffered messages before shutdown, creating a delay to the termination and restart of collector pods. With this update, Fluentd no longer flushes buffers at shutdown, resolving the issue. (link:https://issues.redhat.com/browse/LOG-1739[LOG-1739])
```

File: logging/cluster-logging-release-notes.adoc [hunk 15]
```diff
@@ -192,7 +525,330 @@ In OpenShift Logging 5.3, the legacy configuration mechanism for log forwarding
...
+
+* Before this update, an issue in the bundle manifests prevented installation of the Elasticsearch operator through OLM on OpenShift 4.9. With this update, a correction to bundle manifests re-enables installs and upgrades in 4.9.(link:https://issues.redhat.com/browse/LOG-1780[LOG-1780])
```
Comment (kalexand-rh) [inline logging/cluster-logging-release-notes.adoc:820]:
```suggestion
* Before this update, an issue in the bundle manifests prevented installation of the Elasticsearch Operator through OLM on {product-title} 4.9. With this update, a correction to bundle manifests re-enables installation and upgrade in 4.9.(link:https://issues.redhat.com/browse/LOG-1780[LOG-1780])
```

> The term "Operator" is always capitalized

File: modules/cluster-logging-loki-tech-preview.adoc [hunk 16]
```diff
@@ -7,15 +7,15 @@
 include::snippets/technology-preview.adoc[]
 
 [id="cluster-logging-about-loki"]
-= About Loki
+== About Loki
```
Comment (kalexand-rh) [inline modules/cluster-logging-loki-tech-preview.adoc:10]: The first heading in a modules must be H1, so you need to revert this change and adjust the offset in the assembly.

All of the headings in the following modules need to be adjusted.

**Review by kalexand-rh — COMMENTED** [review]
There are a couple of style and formatting issues. I pointed out the first instance of several of them, but please double-check the object markup and presence of the object type after object names throughout.

File: logging/cluster-logging-release-notes.adoc [hunk 17]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+[id="openshift-logging-5-4-1-bug-fixes"]
+=== Bug fixes
+* Before this update, the log file metric exporter only reported logs created while the exporter was running, which resulted in inaccurate log growth data. This update resolves this issue by monitoring `/var/log/pods`. (https://issues.redhat.com/browse/LOG-2442[LOG-2442])
+
+* Before this update, the collector would be blocked because it continually tried to use a stale connection when forwarding logs to fluentd forward receivers. With this release, the `keepalive_timeout` value has been set to 30 seconds (`30s`) so that the collector recycles the connection and re-attempts to send failed messages within a reasonable amount of time. (https://issues.redhat.com/browse/LOG-2534[LOG-2534])
+
+* Before this update, an error in the gateway component enforcing tenancy for reading logs limited access to logs with a Kubernetes namespace causing "audit" and some "infrastructure" logs to be unreadable. With this update, the proxy correctly detects users with admin access and allows access to logs without a namespace. (https://issues.redhat.com/browse/LOG-2448[LOG-2448])
```
[the remark this answers is not in the document]
Comment (libander) [reply logging/cluster-logging-release-notes.adoc:24]: They're references to types of objects. There is nothing in the markup reference or supplemental style guide that prohibit quotation marks.

**Review by kalexand-rh — COMMENTED** [review]
Please address my original review comments.

**Comment by libander** [comment]
> Please address my original review comments.

They were addressed. https://github.com/openshift/openshift-docs/compare/00e5bd44ec4628d614070746a131f060bff549e0..c174a61ce65b549b3515e72ae762510c91d4c73d

File: logging/cluster-logging-release-notes.adoc (hunk 17 above)
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply logging/cluster-logging-release-notes.adoc:24]: Except it says that system objects are enclosed in backticks.

File: logging/cluster-logging-release-notes.adoc [hunk 18]
```diff
@@ -11,29 +11,362 @@ toc::[]
...
+
+* Before this update, tuning the buffer chunk size could cause the collector to generate a warning about the chunk size exceeding the byte limit for the event stream. With this update, you can also tune the read line limit, resolving the issue.	(link:https://issues.redhat.com/browse/LOG-2379[LOG-2379])
+
+* Before this update, the logging console link in OpenShift web console was not removed with the ClusterLogging CR. With this update, deleting the CR or uninstalling the Cluster Logging Operator removes the link. (link:https://issues.redhat.com/browse/LOG-2373[LOG-2373])
+
+* Before this update, a change to the container logs path caused the collection metric to always be zero with older releases configured with the original path. With this update, the plug-in which exposes metrics about collected logs supports reading from either path to resolve the issue. (link:https://issues.redhat.com/browse/LOG-2462[LOG-2462])
```
[the remark this answers is not in the document]
Comment (libander) [reply logging/cluster-logging-release-notes.adoc:102]: Done. Please merge.
